### PR TITLE
Add axis title standoff

### PR DESCRIPTION
**Pull request by etpinard**
resolves https://github.com/plotly/plotly.js/issues/4224

This PR adds a new attribute `title.standoff` to cartesian axes - which determines the distance (in px) between the tick labels (at max depth) and the axis title.

example 1: https://codepen.io/etpinard/pen/LYYZJXw?editors=0010
example 2: https://codepen.io/etpinard/pen/jOOrvdZ?editors=1010

cc @plotly/plotly_js

### CODE REVIEW

File: src/plots/cartesian/axes.js [hunk 1]
```diff
@@ -2110,6 +2109,10 @@ axes.getTickSigns = function(ax) {
     if((ax.ticks !== 'inside') === (axLetter === 'x')) {
         out = out.map(function(v) { return -v; });
     }
+    // independent of `ticks`; do not flip this one
+    if(ax.side) {
+        out.push({l: -1, t: -1, r: 1, b: 1}[ax.side.charAt(0)]);
```
[the remark this answers is not in the document]
Comment (etpinard) [reply src/plots/cartesian/axes.js]: Can I ask why? `out` is defined as length=4 array

https://github.com/plotly/plotly.js/blob/bdf312ba4f12ce4ea7015f49abc09b2a88a8b5e9/src/plots/cartesian/axes.js#L2107

a couple lines above, and `out[4]` only makes sense when `side` is defined which isn't always the case when `Axes.tickSigns` is called in the indicator, ternary, polar and colorbar code.

File: src/plots/cartesian/axes.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ var MINUS_SIGN = constants.MINUS_SIGN;
 var BADNUM = constants.BADNUM;
 
 var MID_SHIFT = require('../../constants/alignment').MID_SHIFT;
+var CAP_SHIFT = require('../../constants/alignment').CAP_SHIFT;
```
[the remark this answers is not in the document]
Comment (etpinard) [reply src/plots/cartesian/axes.js]: Sure! Done in be3de9293

File: src/plots/cartesian/axes.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (archmoj) [reply src/plots/cartesian/axes.js]: Simply to make sure if one adds another optional item to the list. If he/she copies those lines to push, without `ax.side` `out[5]` would be pushed to `out[4]` location. 
Anyway this is non-blocking.

**Comment by archmoj** [comment]
@etpinard My comment above is non-blocking; 
so please feel free to merge.
:dancer:

**Pull request merged**